tasks/src/tasks/expunge.rs

```diff
@@ -0,0 +1,43 @@
+use imap_types::{
+    command::CommandBody,
+    response::{Data, StatusBody, StatusKind},
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug, Default)]
+pub struct ExpungeTask {
+    count: usize,
```

Comment:
I'm not sure I understand the idea of the count. We expect a list of expunged messages, right?

Reply:
Ah, maybe adding docs will clear things up :-) I also keep forgetting...

Reply:
I initially wanted to return a `Vec<NonEmptyU32>`, but:

> For example, if the last 5 messages in a 9-message mailbox are expunged, a "lower to higher" server will send five untagged EXPUNGE responses for message sequence number 5, whereas a "higher to lower server" will send successive untagged EXPUNGE responses for message sequence numbers 9, 8, 7, 6, and 5.

Because server behaviour is not well defined, this task could not output the same. It could just output a vector with 5 times the same sequence number… so I ended up just counting the number of expunged messages.

Let me add a tiny doc about it.

Reply:
Returning `Vec` allows a consumer to easily do `vec.len()`. But returning `vec.len()` only "removes" potentially useful information. I think we should be careful with this because then it's only a matter of time someone will need to redefine the task (or add another one).

Luckily, it's easy to define custom `Task`s. So, maybe we should try to be a bit more generic for the build-in ones and do more opinionated tasks at higher level?

I think we should avoid dropping information that is already there.

Reply:
I added a better comment to be more explicit.